dropshot/src/server.rs

```diff
@@ -73,6 +73,40 @@ pub struct DropshotState<C: ServerContext> {
     /// Worker for the handler_waitgroup associated with this server, allowing
     /// graceful shutdown to wait for all handlers to complete.
     pub(crate) handler_waitgroup_worker: DebugIgnore<waitgroup::Worker>,
+    /// Determines how to construct responses for server errors.
+    // Since most of the complex type here is the `dyn Fn` trait, we can't
+    // easily simplify it by just stuffing it in a type alias --- we'd have to
+    // add our own trait that's implemented for `T: Fn(ErrorContext<'_, C>,
```

Comment:
Out of curiosity, why not just make this a trait object?  That seems like a clean way to let consumers provide what they want here.

Reply:
What do you mean? The current approach is a trait object --- it's a `Box<dyn Fn(ErrorContext<'_>, ServerError) -> Response<Body> + Send + Sync>`. The comment is regarding how `clippy` dislikes the type _of the boxed trait object_; we could give the type a shorter name by introducing a trait with a blanket impl (`impl<T> ErrorHandler for T where T: Fn(...) -> ...)`) and changing the type to `Box<dyn ErrorHandler + Send + Sync>`, but I didn't love the idea of introducing a trait with a blanket impl just to make the clippy warning go away.

Reply:
Sorry, what I meant was defining a user-facing trait called `ErrorResponseSerializer` or something like that with one method which has the same signature as this function and then this would just be `Box<dyn ErrorResponseSerializer>`.  I wouldn't do it for the clippy warning but because it seems a little cleaner.  (I don't feel strongly about it.)

Reply:
Yeah, that was what I was saying I didn't do in the comment. Broadly, my preference has been to _avoid_ such "trait aliases" (traits that are blanket implemented for all `T: SomeOtherTrait`) in public APIs, since I've seen folks get confused a lot by APIs that expect some trait that you can't actually implement and can only implement by implementing another trait. But, I'm open to being convinced otherwise, if you think that it's better!

Reply:
I don't mean a trait alias with a blanket impl for functions with this signature.  I mean that the consumer would use their own `struct MySerializer` and explicitly `impl ErrorResponseSerializer for MySerializer` and impl the trait's function and then they'd pass that struct into the builder that constructs the Dropshot server.

Reply:
Oh, I see what you mean. Hm, yeah, that could also work.